pyproject.toml

```diff
@@ -69,6 +69,36 @@ mkdocs-material = "^5.5.5"
 # HACK https://readthedocs.org/api/v2/build/11653917.txt
 # TODO Remove this dependency when mkdocstrings supports pymdown-extensions==8.0
 pymdown-extensions = "<8.0,>=6.3"
+tox = "^3.19.0"
+
+[tool.poe.tasks]
+clean.sequence = ["_clean-build", "_clean-pyc"]
+clean.help = "remove build/python artifacts"
+_clean-build = """
+rm -rf build/
+       dist/
+       *.egg-info
+       pip-wheel-metadata
+       *.egg-info
+"""
+_clean-pyc = """
+rm -rf .pytest_cache
+       ./**/__pycache__
+       ./**/*.pyc
+       ./**/*.pyo
+       ./**/*~
+"""
```

Comment:
I don't think this is very useful because builds are done in CD jobs. Also these would fail on Windows:

```suggestion
```

Reply:
This is meant to match the functionality in the makefile, which I guess wasn't very windows friendly either.

If it's really not wanted I can remove it, though I'll add that in my experience having an easy way to clean up various artefacts from the project sometimes comes in handy, as does occasionally creating full builds as part of the dev workflow.

Reply:
Well, if you add it, please use Python code, so it can be used on Windows. Windows support las only started being tested recently. Otherwise, I prefer to remove it and add it later if needed.

Reply:
My feeling is that putting "code" in a config file isn't great, so it's not directly supported. Still there're currently two ways to make it work:

```toml
_clean-build = """
python -c "import os;\
etc\
"
"""
```

which is not so nice,

Or we could put a separate python module in the project e.g. `./devtasks.py` or what have you and reference functions from it like:

```toml
clean.script = "devtasks:clean"
```

which is clean enough in the toml, but comes with the overhead of another free floating python module in this case.

Also for the sake of discussion: it's fairly easy to add a new task type, I'm even thinking poe should support task types defined as third party plugins, so a plumbum or raw python or whatever task type is conceivable. Though it would take a bit more analysis and development to support.

What do you think?

Reply:
> we could put a separate python module in the project e.g. `./devtasks.py` or what have you and reference functions from it

Seems like a good option!